src/cp.js

```diff
@@ -84,18 +84,12 @@ function copyFileSync(srcFile, destFile, options) {
 //
 // Licensed under the MIT License
 // http://www.opensource.org/licenses/mit-license.php
-function cpdirSyncRecursive(sourceDir, destDir, opts) {
+function cpdirSyncRecursive(sourceDir, destDir, currentDepth, opts) {
   if (!opts) opts = {};
 
-  /* Ensure there is not a run away recursive copy. */
-  if (typeof opts.depth === 'undefined') {
-    opts.depth = 0;
-  }
-  if (opts.depth >= common.config.maxdepth) {
-    // Max depth has been reached, end copy.
-    return;
-  }
-  opts.depth++;
+  // Ensure there is not a run away recursive copy
+  if (currentDepth >= common.config.maxdepth) return;
+  currentDepth++;
```

Comment:
As for the order, I usually leave options as the final parameter. `cpdirSyncRecursive` is totally internal, so requiring the `depth` parameter isn't so awful.

If we allow `depth` and `opts` to both be optional, then we have to account for the case when `depth` is missing but `opts` isn't, and vice-versa. Sounds very bug-prone to me, for very little external benefit.